src/django/api/views/v1/moderation_events.py

```diff
@@ -112,43 +113,60 @@ def partial_update(self, request, pk=None):
             partial=True
         )
 
-        if serializer.is_valid():
-            serializer.save()
-            return Response(serializer.data, status=status.HTTP_200_OK)
+        if not serializer.is_valid():
+            raise ValidationError({
+                "detail": APIV1CommonErrorMessages.COMMON_REQ_BODY_ERROR,
+                "errors": [serializer.errors]
+            })
 
-        return Response(
-            {
-                "detail": 'The request body contains '
-                'invalid or missing fields.',
-                "error": [serializer.errors]
-            },
-            status=status.HTTP_400_BAD_REQUEST
-        )
+        serializer.save()
+
+        return Response(serializer.data, status=status.HTTP_200_OK)
 
     @handle_errors_decorator
-    @action(detail=True, methods=['POST'], url_path='production-locations')
-    def add_production_location(self, request, pk=None):
+    @action(
+        detail=True, methods=['POST', 'PATCH'], url_path='production-locations'
+    )
+    def manage_production_location(self, request, pk=None):
         ModerationEventsService.validate_user_permissions(request)
 
         ModerationEventsService.validate_uuid(pk)
 
-        event = ModerationEventsService.fetch_moderation_event_by_uuid(
-            pk
-        )
-
+        event = ModerationEventsService.fetch_moderation_event_by_uuid(pk)
         ModerationEventsService.validate_moderation_status(event.status)
 
-        add_production_location = EventApprovalContext(
-            AddProductionLocation(event)
-        )
+        if request.method == 'POST':
+            production_location_processor = AddProductionLocation(event)
+            status_code = status.HTTP_201_CREATED
+
+        else:
+            serializer = UpdateProductionLocationSerializer(data=request.data)
```

Comment:
For this part, I recommend implementing something like this

  @detail_route(methods=['get', 'head'], url_path='foo', url_name='foo')
   def foo(self, request, pk=None):
       if request.method == 'get':
           return self.foo_get(request, pk)
       if request.method == 'head':
           return self.foo_head(request, pk)
   def foo_get(self, request, pk):
       pass
   def foo_head(self, request, pk):
       pass
       
I think it will make the code leaner

Reply:
I refactored the code using [Routing additional HTTP methods for extra actions](https://www.django-rest-framework.org/api-guide/viewsets/#routing-additional-http-methods-for-extra-actions)
cc @VadimKovalenkoSNF